src/Compilers/CSharp/Portable/FlowAnalysis/DataFlowPass.cs

```diff
@@ -271,14 +271,19 @@ protected override ImmutableArray<PendingBranch> RemoveReturns()
 
         protected virtual void ReportUnassignedOutParameter(ParameterSymbol parameter, SyntaxNode node, Location location)
         {
-            if (!_requireOutParamsAssigned && topLevelMethod == currentMethodOrLambda) return;
+            if (!_requireOutParamsAssigned && topLevelMethod == currentMethodOrLambda ||
+                node == null && location == null)
```

Comment:
If `location` is `null` it could be created from `node`, however if both are `null` then there is no point to try validate further. It will lead to a `NullReferenceException`(this causes VS to crash) and a `location` is needed to create diagnostics. I couldn't find another way that both could be `null` since I added a `null `check in DataFlowPass.LocalFunctions.cs that actually prevented the issue already from this bug (created by `PendingBranches.Add(new PendingBranch(null, this.State));`)

I wanted to make sure if on some way it would happen again, its better to catch it than let VS run into crash.